stix2/registration.py

```diff
@@ -3,10 +3,83 @@
 from . import registry
 from .base import _DomainObject, _Observable
 from .exceptions import DuplicateRegistrationError
-from .properties import _validate_type
+from .properties import (
+    _validate_type, ListProperty, ObjectReferenceProperty, ReferenceProperty,
+)
 from .utils import PREFIX_21_REGEX, get_class_hierarchy_names
 from .version import DEFAULT_VERSION
 
+# Properties ending in "_ref/s" need to be instances of specific types to meet
+# our interpretation of section 3.1 of the spec
+_VERSION_REF_TYPES_MAP = {
+    # TODO: The 2.0 code seems to use `ReferenceProperty`s - is that okay?
```

Comment:
> The 2.0 code seems to use `ReferenceProperty`s - is that okay?

rather than `ObjectReferenceProperty`s. AFAICT `ReferenceProperty` just seems to be a better version of the other? Is there some spec related reason why it's not valid in 2.0 documents?